contracts-periphery/src/UmbraBatchSend.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+interface IUmbra {
+  function sendEth(
+    address payable receiver,
+    uint256 tollCommitment,
+    bytes32 pkx,
+    bytes32 ciphertext
+  ) external payable;
+
+  function sendToken(
+    address receiver,
+    address tokenAddr,
+    uint256 amount,
+    bytes32 pkx,
+    bytes32 ciphertext
+  ) external payable;
+}
+
+contract UmbraBatchSend {
+  IUmbra internal constant umbra = IUmbra(0xFb2dc580Eed955B528407b4d36FfaFe3da685401);
+
+  struct SendEth {
+    address payable receiver;
+    uint256 amount;
```

Comment:
One thing we should document is that this is the total amount, i.e. amount being sent + toll. 

We should also use `forge-std` to overwrite the toll, and make sure all tests pass with both a zero toll and nonzero toll

Reply:
A good pattern for this is define all the tests in an `abstract` contract, then have two contracts that inherit it: one that sets a zero toll in `setUp`, the other that sets a nonzero toll in `setUp`. Both contracts will then run all the tests that the abstract contract has

Reply:
> One thing we should document is that this is the total amount, i.e. amount being sent + toll.

As I read the code below, this is *not* the total amount. Am I wrong?

Reply:
Right, sorry! Either way should make sure it's clear that total amount paid by user is `sum(_params.amounts) + tollCommitment * _params.length` (and similar docs for each method)

Reply:
@garyghayrat just bumping this docs request